borsh-rs/borsh/src/lib.rs

```diff
@@ -1,11 +1,91 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+#[cfg(feature = "std")]
+extern crate std as alloc;
+
 pub use borsh_derive::{BorshDeserialize, BorshSchema, BorshSerialize};
 
 pub mod de;
 pub mod schema;
 pub mod schema_helpers;
 pub mod ser;
+pub mod error;
+mod write;
 
 pub use de::BorshDeserialize;
 pub use schema::BorshSchema;
 pub use schema_helpers::{try_from_slice_with_schema, try_to_vec_with_schema};
 pub use ser::BorshSerialize;
+
+/// A facade around all the types we need from the `std`, `core`, and `alloc`
+/// crates. This avoids elaborate import wrangling having to happen in every
+/// module.
+#[cfg(feature = "std")]
+pub mod lib {
+    pub use std::{cmp, iter, mem, num, slice, str};
+    pub use std::{f32, f64};
+    pub use std::{i16, i32, i64, i8, isize};
+    pub use std::{u16, u32, u64, u8, usize};
+
+    pub use std::cell::{Cell, RefCell};
+    pub use std::clone::{self, Clone};
+    pub use std::convert::{self, From, Into};
+    pub use std::default::{self, Default};
+    pub use std::fmt::{self, Debug, Display};
+    pub use std::marker::{self, PhantomData};
+    pub use std::ops::Range;
+    pub use std::option::{self, Option};
+    pub use std::result::{self, Result};
```

Comment:
Do we need them? They are available if just imported from `core::` even in std context. I think we only need this facade for third-party collections

Reply:
fixed